.github/workflows/build.yml

```diff
@@ -0,0 +1,430 @@
+        run: brew install ccache
+      - name: Install dependency
+        run: |
+          python3 -m pip install --break-system-packages mako
+      - name: Zero ccache stats
+        run: ccache -z
+      - name: Generate build files with cmake
+        run: |
+          time cmake -DCMAKE_BUILD_TYPE=Debug \
+            -DCMAKE_C_COMPILER_LAUNCHER="cmake;-E;time;ccache" \
+            -DCMAKE_CXX_COMPILER_LAUNCHER="cmake;-E;time;ccache" \
+            -DCMAKE_C_LINKER_LAUNCHER="cmake;-E;time" \
+            -DCMAKE_CXX_LINKER_LAUNCHER="cmake;-E;time" \
+            -GNinja \
+            -Bbuild .
+      - name: Build Dive host tools with ninja
+        run: |
+          time ninja -C build -d stats
+      - name: Show ccache stats
+        run: ccache -s
+      - name: Ninja log
+        run: |
+          cat build/.ninja_log
+      - name: Run tests
+        run: |
+          ctest --output-on-failure -C Debug --test-dir build
+      - name: Upload cache
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ runner.os }}-Release-ccache-continuous-${{ github.sha }}
+
+  build_MacOS_Release:
+    name: "MacOS release build"
+    runs-on: macos-latest
+    env:
+      CCACHE_DIR: ${{ github.workspace }}/.ccache
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install Qt 5.15 via Homebrew
+        run: |
+          brew install qt@5
+          # Link it so standard commands might find it, or just set env vars below
+          echo "Qt5_DIR=$(brew --prefix qt@5)/lib/cmake/Qt5" >> $GITHUB_ENV
+          echo "PATH=$(brew --prefix qt@5)/bin:$PATH" >> $GITHUB_ENV
+      - name: Check Qt Version
+        run: qmake --version
+      - name: Install ccache
+        run: brew install ccache
+      - name: Install dependency
+        run: |
+          python3 -m pip install --break-system-packages mako
+      - name: Zero ccache stats
+        run: ccache -z
+      - name: Generate build files with cmake
+        run: |
+          time cmake -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_C_COMPILER_LAUNCHER="cmake;-E;time;ccache" \
+            -DCMAKE_CXX_COMPILER_LAUNCHER="cmake;-E;time;ccache" \
+            -DCMAKE_C_LINKER_LAUNCHER="cmake;-E;time" \
+            -DCMAKE_CXX_LINKER_LAUNCHER="cmake;-E;time" \
+            -GNinja \
+            -Bbuild .
+      - name: Build Dive host tools with ninja
+        run: |
+          time ninja -C build -d stats
+      - name: Show ccache stats
+        run: ccache -s
+      - name: Ninja log
+        run: |
+          cat build/.ninja_log
+      - name: Run tests
+        run: |
+          ctest --output-on-failure -C Release --test-dir build
+      - name: Upload cache
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ runner.os }}-Debug-ccache-continuous-${{ github.sha }}
```

Comment:
similar as above, is this supposed to be `Release` since it is under `build_MacOS_Release`?